src/version.c

```diff
@@ -98,10 +43,10 @@ version_parse(const char *version, VersionInfo *result)
 	 * the bytes of the string were parsable using the above grammar.
 	 */
 	fields_parsed = sscanf(version, "%lu%n.%lu%n.%lu%n-%" STR(VERSION_INFO_LEN) "s%n",
```

Comment:
can't we simplify this a bunch? we def don't need the %n now? and that let's us get rid of parse_length.

Reply:
So, I think to resolve all your comments on this PR, we should decide how much we want to "validate" what's coming from the server.